### PR TITLE
Reorganize Developing.md sections (fix for #3263)

### PR DESCRIPTION
**Pull request by AlexReiff**
Merged Requirements and Setup Environment sections. Moved new section above Checkout Your Fork because they are prerequisites to it.

# Changes

Reorganized sections of Developing.md to fix the two issues in #3263 so all information needed to setup the development environment and required installations is in one section in the doc. Moved it before the Checkout your Fork section, because that section relies on GOPATH already being configured.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [n/a] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Release notes block has been filled in or deleted (only if no user facing changes)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -45,6 +45,86 @@ At this point, you may find it useful to return to these `Tekton Pipeline` docs:
...
     they are run
+    
+## Environment Setup
+
+You must install these tools:
+
+1.  [`git`](https://help.github.com/articles/set-up-git/): For source control
+
+1.  [`go`](https://golang.org/doc/install): The language Tekton Pipelines is
+    built in
```
Comment (popcor255) [inline DEVELOPMENT.md:56]:
```suggestion
1.  [`git`](https://help.github.com/articles/set-up-git/): For source control

2.  [`go`](https://golang.org/doc/install): The language Tekton Pipelines is
    built in
```

Comment (popcor255) [reply DEVELOPMENT.md:56]: Could you also specify which version of go is needed? I know that if you dont install the correct version. KO will throw a very unhelpful error.

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -45,6 +45,86 @@ At this point, you may find it useful to return to these `Tekton Pipeline` docs:
...
     Define `Tasks`, `Pipelines`, and `PipelineResources`, see what happens when
     they are run
+    
+## Environment Setup
+
+You must install these tools:
+
+1.  [`git`](https://help.github.com/articles/set-up-git/): For source control
+
+1.  [`go`](https://golang.org/doc/install): The language Tekton Pipelines is
```
Comment (popcor255) [inline DEVELOPMENT.md:55]: Could you specify which version of golang is needed to be installed? If you install an old version. I believe that KO throws a very unhelpful error.

Comment (popcor255) [reply DEVELOPMENT.md:55]: Also, I think your need to edit your pr to include `/kind documentation`.

Comment (bobcatfish) [reply DEVELOPMENT.md:55]: just a side note that @AlexReiff was moving the existing docs around - adding the version is a good idea but i think it is np if he doesn't want to be the one to add it

File: DEVELOPMENT.md [hunk 3]
```diff
@@ -6,8 +6,8 @@
 1.  Create [a GitHub account](https://github.com/join)
 1.  Setup
     [GitHub access via SSH](https://help.github.com/articles/connecting-to-github-with-ssh/)
+1.  Set up your [development environment](#environment-setup)
 1.  [Create and checkout a repo fork](#checkout-your-fork)
-1.  Set up your [shell environment](#environment-setup)
 1.  Install [requirements](#requirements)
```
Comment (bobcatfish) [inline DEVELOPMENT.md:11]: i think you deleted this section so maybe we should remove this also?

**Review by bobcatfish — COMMENTED** [review]
Just one nit about removing a link to the section that was deleted

thanks for this @AlexReiff !

**Comment by bobcatfish** [comment]
image

hm looks like this is stuck again, opened https://github.com/tektoncd/plumbing/issues/581

File: DEVELOPMENT.md (hunk 3 above)
Comment (AlexReiff) [reply DEVELOPMENT.md]: Ahh good catch, updated.

File: DEVELOPMENT.md (hunk 2 above)
Comment (AlexReiff) [reply DEVELOPMENT.md:54]: I debated this. I decided against this because it's more an issue with installing Ko than installing Tekton.

**Comment by popcor255** [comment]
Thanks for this! 😸 
/meow

**Pull request merged**